### DISCUSSION TITLE
What is the storage logic of es service/instance/endpoind traffic?

### DISCUSSION THREAD

**Discussion by huangweikuna** [apache/skywalking#9588] [Q&A] [answered]
hi, skywalking team.

## Background
I've been working on a project to convert the metrics of Skywalking to prometheus (I've read that the data of sw will burst the data of prometheus, but here we don't need all the data, only some of it), after finishing the conversion of service and instance metrics, everything is going smoothly and the amount of data is under control, but a problem will However, there is a problem that we need to label service/instance when converting the data into PROM, my approach is to query all the data of sw_service_traffic and sw_instance_traffic, establish the correspondence of id -> name, and then get the corresponding ID in the indicator for conversion. Then, we can get the corresponding ID in the indicator and transform it. Here there will be a situation that the corresponding service and instance cannot be found, in general.
Example.
The corresponding record in _index -> sw_metrics-sum-20220909 has service_id -> c2l0LXphYmFuay1zYnMtdGljay1zZXJ2aWNl.1
is not found in the above correspondence, which will lead to the failure of the metric transformation.
The instance_id also has a problem with the correspondence. Am I overlooking something?

## Temporarily

We ignored some of the metrics generation, but recently we need to do the endpoint metrics, what stops people is that the data volume of sw_endpoint_traffic is too big, I tried to query only the data of sw_endpoint_traffic-${today}, but it failed, the metrics that query the data will appear in the index of 3 days ago This makes it necessary to query all the data, can this be avoided?

hi，skywalking 团队：

## 背景
最近在做一个把Skywalking的指标转化为prometheus的指标（我看过相映的调查，sw的数据会把prom的撑爆，但这里我们并不是需要所有的数据，只需要部分）在完成service和instance的指标转化之后，一切都很顺利，数据量也在可控范围，但是会出现一个问题，在转化为prom数据的时候，需要打上service/instance的标签，我的做法是查询出所有 sw_service_traffic、sw_instance_traffic的数据，建立 id -> name的对应关系，然后在指标中取到对应的ID进行转化。这里会出现找不到对应的service和instance的情况，总的来说就是：
例如：
在 _index -> sw_metrics-sum-20220909 中对应的记录的 service_id -> c2l0LXphYmFuay1zYnMtdGljay1zZXJ2aWNl.1
在上述对应关系中没办法找到，这会导致指标转化的失败。
instance_id也会有对应的问题。是我忽略了哪里嘛？

## 暂时

我们忽略了部分指标的生成，但是在最近需要做endpoint的指标，让人止步的是，sw_endpoint_traffic的数据量太大了，我尝试了只查询 sw_endpoint_traffic-${today} 的数据，但是失败了，查询到数据的指标会出现在 3天前的索引，这使得还是不得不查询所有的数据，这个能避免嘛？

**Comment by wu-sheng** [comment, marked as answer]
endpoint is the reason why we said repeatedly, don't try to push skywalking data into Prometheus.

I do think there is nothing we can do. The reality is APM has to do more than that. All commercial APMs have the same case.

**Reply by huangweikuna** [reply]
ok, thanks for your replay
